Sources/KlaviyoCore/Models/APIModels/FullForms.swift

```diff
@@ -0,0 +1,48 @@
+//
+//  FullForms.swift
+//  klaviyo-swift-sdk
+//
+//  Created by Andrew Balmer on 11/25/24.
+//
+
+import Foundation
+
+public struct FullForms: Equatable {
+    public let forms: [Data]
+    public let formSettings: Data
+    public let dynamicInfoConfig: Data?
+
+    private enum CodingKeys: String, CodingKey {
+        case forms = "full_forms"
+        case formSettings = "form_settings"
+        case dynamicInfoConfig = "dynamic_info_config"
+    }
+
+    public init(data: Data) throws {
+        guard JSONSerialization.isValidJSONObject(data) else {
```

Comment:
This seems a little odd to me for an initializer. Is there a reason we are doing all this serialization/deserialization in here. You going to use it in another pr?

Reply:
I'm trying to mimic a custom JSON decoder here, like [this](https://www.donnywals.com/writing-custom-json-encoding-and-decoding-logic/). The `FullForms` struct isn't actually `Decodable`, though, so I can't use the `init(from decoder: Decoder) throws { ... }` method, so I implemented my own similar initializer that takes in the raw `Data` type, extracts the top-level values, and stores them as `Data`. Notice that I'm only using static `JSONSerialization` methods here; I'm not actually using any instance of `JSONDecoder`. 

I had hoped that I could just conform the `FullForms` object to `Codable`, and the default JSONDecoder would see that I want to decode into `Data` types, then do it for me automatically. I tried getting that to work but Swift doesn't like that; seems like it wants me to decode the `Data` into concrete `Swift` types.

Reply:
Given the [conversation with @ajaysubra](https://github.com/klaviyo/klaviyo-swift-sdk/pull/240#discussion_r1859316460) above, though, I am wondering if I should just decode all of the response `Data` into concrete Swift types, so that I can get `Codable` conformance automatically and avoid doing this custom serialization work. We may end up re-serializing the data in order to pass into the javascript, but maybe that's not so bad. And we may end up wanting to do something with the decoded forms response down the line, even if it's post-MVP

Reply:
Ok I modified the `FullForms` struct to make it `Codable`. Now we can decode it using JSONDecoder. For now, the structure is a skeleton, but we'll fill in the properties that we need once we figure out what's necessary in order to drive the webview.